generate-certs/main.tf

```diff
@@ -0,0 +1,155 @@
+      "Principal": {
+         "Service": "ec2.amazonaws.com"
+      }
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "s3_gen_certs" {
+  name = "s3_gen_certs"
+  role = "${aws_iam_role.gen_certs_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:Get*",
+        "s3:PutObject"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${replace(var.s3_root_path,"/(\/)+$/","")}",
+        "arn:aws:s3:::${replace(var.s3_root_path,"/(\/)+$/","")}/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:List*"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${element(split("/", var.s3_root_path), 0)}"
+      ]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "gen_certs_tags" {
+  name = "gen-certs-tags"
+  role = "${aws_iam_role.gen_certs_role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+          "ec2:CreateTags",
+          "ec2:AssociateAddress",
+          "ec2:DescribeAddresses",
+          "ec2:DescribeInstances"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_instance_profile" "gen_certs_profile" {
+  name  = "${var.stack_item_label}-${var.region}-gen-certs"
+  roles = ["${aws_iam_role.gen_certs_role.name}"]
+}
+
+/* ---------------------------- */
+/* Security Group               */
+/* ---------------------------- */
+resource "aws_security_group" "generate_certs_sg" {
+  name        = "${var.stack_item_label}-${var.region}-gen-certs-sg"
+  description = "${stack_item_fullname} security group"
+}
+
+resource "aws_security_group_rule" "allow_ssh_in_tcp" {
+  type        = "ingress"
+  from_port   = 22
+  to_port     = 22
+  protocol    = "tcp"
+  cidr_blocks = ["${split(",",var.cidr_whitelist)}"]
+
+  security_group_id = "${aws_security_group.generate_certs_sg.id}"
+}
+
+/* ---------------------------- */
+/* Instance & User Data         */
+/* ---------------------------- */
+resource "template_file" "user_data" {
+  template = "${file("${path.module}/templates/user_data.tpl")}"
+
+  vars {
+    s3_region         = "${var.region}"
+    s3_cert_root_path = "s3://${var.s3_root_path}"
+    key_size          = "${var.cert_key_size}"
+    s3_dir_override   = "${var.s3_dir_override}"
+    key_city          = "${var.key_city}"
+    key_org           = "${var.key_org}"
+    key_email         = "${var.key_email}"
+    key_ou            = "${var.key_ou}"
+    cert_key_name     = "${var.cert_key_name}"
+    key_country       = "${var.key_country}"
+    key_province      = "${var.key_province}"
+    active_clients    = "${var.active_clients}"
+    revoked_clients   = "${var.revoked_clients}"
+    openvpn_host      = "${var.openvpn_host}"
+    force_cert_regen  = "${var.force_cert_regen}"
+    s3_push_dryrun    = "${var.s3_push_dryrun}"
+  }
+}
+
+resource "aws_instance" "generate_certs" {
```

Comment:
kept it simple, since this node doesn't need to be in a vpc, or event an ec2 node for that matter 🙊